### PR TITLE
Samplings deprecation and shift to spharpy

### PR DESCRIPTION
**Pull request by ahms5**
As discussed earlier we want to shift the samplings to spharpy. therefore we deprecate it now.
https://github.com/pyfar/spharpy/pull/33

### CODE REVIEW

File: examples/pyfar_demo.ipynb [hunk 1]
```diff
@@ -475,7 +475,7 @@
    "source": [
     "## Entering coordinate points<a class=\"anchor\" id=\"coordinates_enter\"></a>\n",
     "\n",
-    "Coordinate points can be entered manually or by using one of the available sampling schemes contained in `pyfar.samplings`. We will do the latter using an equal angle sampling and look at the information provided by the coordinates object:"
+    "Coordinate points can be entered manually."
```
Comment (f-brinkmann) [inline examples/pyfar_demo.ipynb:478]: Maybe also add a note that samplings are available from spharpy?

**Review by f-brinkmann — DISMISSED** [review]
Looks good to me. Only one minor comment, that could be included

File: examples/pyfar_demo.ipynb (hunk 1 above)
Comment (ahms5) [reply examples/pyfar_demo.ipynb:478]: I think we need a general overview, where to find what. Maybe the website would be a good place.

File: pyfar/classes/coordinates.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 The following documents the pyfar coordinates class and functions for
 coordinate conversion. More background information is given in
 :py:mod:`coordinates concepts <pyfar._concepts.coordinates>`.
-Available sampling schemes are listed at :py:mod:`~pyfar.samplings`.
+Available sampling schemes are listed at :py:mod:`~spharpy.samplings`.
```
Comment (mberz) [inline pyfar/classes/coordinates.py:5]: For external package documentations to be linked, you'll need to specify these more explicitly.
There's the `intersphinx` plug-in for this. I've used it in spharpy to reference the Pyfar docs there. You can have a look at the respective config in spharpy

for example in spharpy, I used 
:doc:`pf.Coordinates <pyfar:classes/pyfar.coordinates>`
to reference the pyfar Coordinates.

File: pyfar/samplings/samplings.py [hunk 3]
```diff
@@ -13,6 +14,9 @@ def cart_equidistant_cube(n_points):
     """
     Create a cuboid sampling with equidistant spacings in x, y, and z.
 
+    This function will be deprecated in pyfar 0.8.0 in favor
+    of :py:class:`spharpy.samplings.cube_equidistant`.
```
Comment (mberz) [inline pyfar/samplings/samplings.py:18]: see comment above

File: pyfar/samplings/samplings.py [hunk 4]
```diff
@@ -61,6 +70,9 @@ def sph_dodecahedron(radius=1.):
     Generate a sampling based on the center points of the twelve
     dodecahedron faces.
 
+    This function will be deprecated in pyfar 0.8.0 in favor
+    of :py:class:`spharpy.samplings.dodecahedron`.
```
Comment (mberz) [inline pyfar/samplings/samplings.py:74]: see comment above

File: pyfar/samplings/samplings.py [hunk 5]
```diff
@@ -116,6 +131,9 @@ def sph_icosahedron(radius=1.):
     """
     Generate a sampling from the center points of the twenty icosahedron faces.
 
+    This function will be deprecated in pyfar 0.8.0 in favor
+    of :py:class:`spharpy.samplings.icosahedron`.
```
Comment (mberz) [inline pyfar/samplings/samplings.py:135]: see comment above

File: pyfar/samplings/samplings.py [hunk 6]
```diff
@@ -150,6 +173,9 @@ def sph_equiangular(n_points=None, sh_order=None, radius=1.):
     """
     Generate an equiangular sampling of the sphere.
 
+    This function will be deprecated in pyfar 0.8.0 in favor
+    of :py:class:`spharpy.samplings.equiangular`.
```
Comment (mberz) [inline pyfar/samplings/samplings.py:177]: see comment above

File: pyfar/samplings/samplings.py [hunk 7]
```diff
@@ -234,6 +265,9 @@ def sph_gaussian(n_points=None, sh_order=None, radius=1.):
     """
     Generate sampling of the sphere based on the Gaussian quadrature.
 
+    This function will be deprecated in pyfar 0.8.0 in favor
+    of :py:class:`spharpy.samplings.gaussian`.
```
Comment (mberz) [inline pyfar/samplings/samplings.py:269]: see comment above

File: pyfar/samplings/samplings.py [hunk 8]
```diff
@@ -313,6 +352,9 @@ def sph_extremal(n_points=None, sh_order=None, radius=1.):
     """
     Return a Hyperinterpolation sampling grid.
 
+    This function will be deprecated in pyfar 0.8.0 in favor
+    of :py:class:`spharpy.samplings.hyperinterpolation`.
```
Comment (mberz) [inline pyfar/samplings/samplings.py:356]: see comment above

File: pyfar/samplings/samplings.py [hunk 9]
```diff
@@ -407,6 +453,9 @@ def sph_t_design(degree=None, sh_order=None, criterion='const_energy',
     """
     Return spherical t-design sampling grid.
 
+    This function will be deprecated in pyfar 0.8.0 in favor
+    of :py:class:`spharpy.samplings.spherical_t_design`.
```
Comment (mberz) [inline pyfar/samplings/samplings.py:457]: see comment above

File: pyfar/samplings/samplings.py [hunk 10]
```diff
@@ -537,6 +590,9 @@ def sph_equal_angle(delta_angles, radius=1.):
     """
     Generate sampling of the sphere with equally spaced angles.
 
+    This function will be deprecated in pyfar 0.8.0 in favor
+    of :py:class:`spharpy.samplings.equal_angle`.
```
Comment (mberz) [inline pyfar/samplings/samplings.py:594]: see comment above

File: pyfar/samplings/samplings.py [hunk 11]
```diff
@@ -601,6 +661,9 @@ def sph_great_circle(elevation=np.linspace(-90, 90, 19), gcd=10, radius=1,
     """
     Spherical sampling grid according to the great circle distance criterion.
 
+    This function will be deprecated in pyfar 0.8.0 in favor
+    of :py:class:`spharpy.samplings.great_circle`.
```
Comment (mberz) [inline pyfar/samplings/samplings.py:665]: see comment above

File: pyfar/samplings/samplings.py [hunk 12]
```diff
@@ -689,8 +755,11 @@ def sph_lebedev(n_points=None, sh_order=None, radius=1.):
     """
     Return Lebedev spherical sampling grid.
 
+    This function will be deprecated in pyfar 0.8.0 in favor
+    of :py:class:`spharpy.samplings.lebedev`.
```
Comment (mberz) [inline pyfar/samplings/samplings.py:759]: see comment above

File: pyfar/samplings/samplings.py [hunk 13]
```diff
@@ -689,8 +755,11 @@ def sph_lebedev(n_points=None, sh_order=None, radius=1.):
...
+
     For detailed information, see [#]_. For a list of available values
-    for `n_points` and `sh_order` call :py:func:`sph_lebedev`.
+    for `n_points` and `sh_order` call :py:func:`lebedev`.
```
Comment (mberz) [inline pyfar/samplings/samplings.py:762]: see comment above

File: pyfar/samplings/samplings.py [hunk 14]
```diff
@@ -786,6 +859,9 @@ def sph_fliege(n_points=None, sh_order=None, radius=1.):
     """
     Return Fliege-Maier spherical sampling grid.
 
+    This function will be deprecated in pyfar 0.8.0 in favor
+    of :py:class:`spharpy.samplings.fliege`.
```
Comment (mberz) [inline pyfar/samplings/samplings.py:863]: see comment above

File: pyfar/samplings/samplings.py [hunk 15]
```diff
@@ -950,6 +1029,9 @@ def sph_equal_area(n_points, radius=1.):
     """
     Sampling based on partitioning into faces with equal area.
 
+    This function will be deprecated in pyfar 0.8.0 in favor
+    of :py:class:`spharpy.samplings.equal_area`.
```
Comment (mberz) [inline pyfar/samplings/samplings.py:1033]: see comment above

File: examples/pyfar_demo.ipynb (hunk 1 above)
Comment (mberz) [reply examples/pyfar_demo.ipynb:478]: Using links in the respective documentations will help to some degree, but a short description on the website surely does help. It does not need to be a full listing of all functionality, that's part of the docs.

**Review by mberz — CHANGES_REQUESTED** [review]
Please use intersphinx to link between the docs of different packages.

**Review by f-brinkmann — APPROVED** [review]

**Review by mberz — APPROVED** [review]

**Pull request merged**